### ISSUE TITLE
Refactor-nrepl middleware not enabled because Cider cannot find project

### ISSUE THREAD

**Issue by dmg46664** [clojurephant/clojurephant#131]
## Desired Behavior

I've raised this as a feature request on the assumption that it isn't present.
I'm new to all this so it may be that I don't understand precisely what I'm doing.

The desired behaviour is for the following to go away.

```
WARNING: No Clojure project was detected. The
refactor-nrepl middleware was not enabled. (You can mute this
warning by customizing `cljr-suppress-no-project-warning'.)
user> 
```

## Context

I have the cider-repl and handler connected

```
implementation 'cider:cider-nrepl:0.23.0-SNAPSHOT'

...

(defn nrepl-handler []
  (require 'cider.nrepl)
  (ns-resolve 'cider.nrepl 'cider-nrepl-handler))

(defn start-server [port]
  (nrepl-server/start-server :port port :handler (nrepl-handler)))
```

I connect fine but get the above mentioned warning. I presume that the refactoring tooling is trying to parse something locally about the project but cannot see it because of some mismatched expectation.

**Comment by dmg46664** [comment]
It's entirely possible that Cider has to support clojurephant, which seems a pity if that's the way it works.

**Comment by dmg46664** [comment]
Hmmm seems to be clj-refactor check
https://github.com/clojure-emacs/clj-refactor.el/blob/master/clj-refactor.el#L3242

**Comment by dmg46664** [comment]
If everything above does make sense, then I think `clojurephant` should have a not-supported features section on the main page, to let Emacs users know what they will sacrifice if they choose this build tool.

Presumably this also means hot-loading dependencies in the REPL is not supported https://practicalli.github.io/spacemacs/refactor/clj-refactor/managing-library-dependencies.html

**Comment by ajoberstar** [comment]
Do you have a dependency on `refactor-nrepl`?

**Comment by dmg46664** [comment]
@ajoberstar Not in my project, but enabled in Emacs I believe . If I understand your question correctly which I might not given I don't understand the difference between `clj-refactor` and `refactor-nrepl`.

I have done the following:
http://develop.spacemacs.org/layers/+lang/clojure/README.html#enabling-sayid-or-clj-refactor

Thanks for replying.

**Comment by dmg46664** [comment]
@ajoberstar - Ahh! Thanks for highlighting this to me!

> On the other hand if a standalone REPL or an embedded nREPL server is used you will need to manually add this dependency (see below).

from  https://github.com/clojure-emacs/clj-refactor.el

However, even though I've added the dependency to `refactor-nrepl`, I believe it needs to be setup as middleware some how? I'm not sure how to do this.

I've sought help here: https://github.com/clojure-emacs/cider/issues/2511#issuecomment-548017961

**Comment by ajoberstar** [comment]
Are you following the instructions in [the FAQ](https://clojurephant.dev/clojurephant/faq.html)? And starting the repl from the `clojureRepl` task? Not sure if this will make a difference.

**Comment by ska2342** [comment]
Cider has support for Gradle. See e.g. https://github.com/clojure-emacs/cider/blob/master/cider.el#L199 
Doesn't work with the Gradle wrapper, though. I'd think that that would be a more useful Gradle support over in Cider since using the wrapper is super standard.

**Comment by ajoberstar** [comment]
I expect this to be fixed by clojure-emacs/cider#3228 and an upgrade to Clojurephant 0.7.0-alpha.6.

**Comment by ajoberstar** [comment]
The cider changes have been merged to master and should be in Cider's next release.
